generators/spring-controller/index.js

```diff
@@ -0,0 +1,101 @@
+const constants = require('../generator-constants');
+const prompts = require('./prompts');
+
+const SERVER_MAIN_SRC_DIR = constants.SERVER_MAIN_SRC_DIR;
+const SERVER_TEST_SRC_DIR = constants.SERVER_TEST_SRC_DIR;
+
+module.exports = class extends BaseGenerator {
+    constructor(args, opts) {
+        super(args, opts);
+        this.argument('name', { type: String, required: true });
+        this.name = this.options.name;
+    }
+
+    initializing() {
+        this.log(`The spring-controller ${this.name} is being created.`);
+        this.baseName = this.config.get('baseName');
+        this.packageName = this.config.get('packageName');
+        this.packageFolder = this.config.get('packageFolder');
+        this.databaseType = this.config.get('databaseType');
+        this.controllerActions = [];
+    }
+
+    get prompting() {
+        return {
+            askForControllerActions: prompts.askForControllerActions
+        };
+    }
+
+    get default() {
+        return {
+            insight() {
+                const insight = this.insight();
+                insight.trackWithEvent('generator', 'spring-controller');
+            }
+        };
+    }
+
+    writing() {
+        this.controllerClass = _.upperFirst(this.name);
+        this.controllerInstance = _.lowerFirst(this.name);
+        this.apiPrefix = _.kebabCase(this.name);
+
+        if (this.controllerActions.length === 0) {
+            this.log(chalk.green('No controller actions found, addin a default action'));
+            this.controllerActions.push({
+                actionName: 'defaultAction',
+                actionMethod: 'Get'
+            });
+        }
+
+        // helper for Java imports
+        this.usedMethods = _.uniq(this.controllerActions.map(action => action.actionMethod));
+        this.usedMethods = this.usedMethods.sort();
+
+        this.mappingImports = this.usedMethods.map(method => `org.springframework.web.bind.annotation.${method}Mapping`);
+        this.mockRequestImports = this.usedMethods.map(method => `static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.${method.toLowerCase()}`);
+
+        // IntelliJ optimizes imports after a certain count
+        this.mockRequestImports = this.mockRequestImports.length > 3 ? ['static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*'] : this.mockRequestImports;
+
+        this.mainClass = this.getMainClassName();
+
+        this.controllerActions.forEach((action) => {
+            action.actionPath = _.kebabCase(action.actionName);
+            action.actionNameUF = _.upperFirst(action.actionName);
+            this.log(chalk.green(`adding ${action.actionMethod} action '${action.actionName}' for /api/${this.apiPrefix}/${action.actionPath}`));
+        });
+
+        this.template(
+            `${SERVER_MAIN_SRC_DIR}package/web/rest/_Controller.java`,
+            `${SERVER_MAIN_SRC_DIR + this.packageFolder}/web/rest/${this.controllerClass}Controller.java`
```

Comment:
this could be `${SERVER_MAIN_SRC_DIR}${this.packageFolder}/web/rest/${this.controllerClass}Controller.java`
same in other usage below